src/main.rs

```diff
@@ -267,17 +268,77 @@ fn build(config: &Config) {
     };
 }
 
+fn static_file_handler(dest: &str, req: Request, res: Response) {
+    let path = PathBuf::from(format!("{}{}", dest, req.uri));
```

Comment:
I was doing exactly this, until I realized it would not work on my machine because `join` was replacing the whole path with `req.uri` rather than appending it. Now that I think about it, my solution was really terrible too. 

I'll experiment a bit here to see how can I grab the string directly.